### ISSUE TITLE
FSlightbox when opening first time throws: Cannot read property 'zero' of undefined

### ISSUE THREAD

**Issue by valerxx22** [banthagroup/fslightbox-react#157]
I have this React component that renders the Lightbox when clicking on an image from gallery page. And seems to work fine overall, but I found a weird behavior, when I clean all my application site data and go the frist time on the gallery page and click an image I get the next error:

> error: uncaughtException: Cannot read property 'zero' of undefined
> TypeError: Cannot read property 'zero' of undefined

> react-hot-loader.development.js:294 TypeError: Cannot read property 'zero' of undefined
>     at t.r.runActions (pathToProject/node_modules/fslightbox-react/index.js:1)
>     at callCallback (pathToProject/node_modules/react-dom/cjs/react-dom.development.js:12490)
>     at commitUpdateQueue (pathToProject/node_modules/react-dom/cjs/react-dom.development.js:12511)
>     at commitLifeCycles (pathToProject/node_modules/react-dom/cjs/react-dom.development.js:19858)
>     at commitLayoutEffects (pathToProject/node_modules/react-dom/cjs/react-dom.development.js:22803)
>     at HTMLUnknownElement.callCallback (pathToProject/node_modules/react-dom/cjs/react-dom.development.js:188)
>     at Object.invokeGuardedCallbackDev (pathToProject/node_modules/react-dom/cjs/react-dom.development.js:237)
>     at invokeGuardedCallback (pathToProject/node_modules/react-dom/cjs/react-dom.development.js:292)
>     at commitRootImpl (pathToProject/node_modules/react-dom/cjs/react-dom.development.js:22541)
>     at unstable_runWithPriority (pathToProject/node_modules/scheduler/cjs/scheduler.development.js:653)

Here is my react component:
```
const Lightbox = () => {
  const dispatch = useDispatch();
  const { items } = useSelector(state => state.items);
  const { lightboxDialog } = useSelector(state => state.window);
  const sources =
    items?.length > 0 ? items.map(item => item.original) : []; // creates an array of strings with url sources to the images

  return (
      <FsLightbox
        toggler={lightboxDialog?.toggler}
        sources={sources}
        slide={lightboxDialog.slide}
      />
  );
};
```

Has anyone had this problem before? How do I fix it since even the stack-trace is pretty hard to follow

**Comment by snelsi** [comment]
Same on my side. It seems to be caused by 'slide' prop.

**Comment by rejas** [comment]
Got this error too when trying to start my lightbox open. At that time the index to open is -1 and that might be the problem.

**Comment by sekizlipenguen** [comment]
Any progress?

**Comment by sekizlipenguen** [comment]
props update issue. Temporary solution could be as follows;
It also only works with "hook".

`      
    const [toggler, setToggler] = useState({index: 0, toggler: false});
    const instagramInfo = Props.instagramInfo;

    let v = false;
    if (instagramItems && instagramItems.length) {
        v = true;
    }

      {
                v && (
                    <FsLightbox
                        toggler={toggler.toggler}
                        sources={instagramItems}
                        sourceIndex={toggler.index}
                    />
                )
            }`

![Screen Shot 2020-09-19 at 13 57 32](https://user-images.githubusercontent.com/36849861/93665560-126b4800-fa80-11ea-9dff-218ab5ef8e3a.png)

**Comment by ryancybul** [comment]
@valerxx22 @rejas @snelsi @sekizlipenguen 

I was running into a similar issue yesterday and was struggling to get it working.  What worked for me was passing the sources array from a parent component into the Lightbox.  Try something like what's listed below.  I hope this works for you!

const Lightbox = ( sourcesArray ) => {
  const dispatch = useDispatch();
  const { items } = useSelector(state => state.items);
  const { lightboxDialog } = useSelector(state => state.window);

  return (
      <FsLightbox
        toggler={lightboxDialog?.toggler}
        sources={sourcesArray}
        slide={lightboxDialog.slide}
      />
  );
};

**Comment by snelsi** [comment]
@ryancybul Sadly, this issue forced me to migrate to [simple-react-lightbox](https://github.com/michelecocuccio/simple-react-lightbox), so it's no longer relevant for me. But thanks anyway.

**Comment by richarddd** [comment]
Just make sure that "sources" is not an empty array and you're good 👍

**Comment by piotrzdziarski** [comment]
As @richarddd noticed it was caused by passing empty **sources**.
This problem was solved in 1.6.0
Now lightbox will detect props changes until first open so it won't crash if you pass no sources at start, but still if you want to update props after first open you need to remount lightbox: https://fslightbox.com/react/documentation/updating-props
